src/support/cleanse.h

```diff
@@ -8,6 +8,8 @@
 
 #include <stdlib.h>
 
+// Securely erase a span of memory in a way guaranteed not to be
```

Comment:
@tjps Nothing in C++ can guarantee that memory will be overwritten. The `memset_s` function from the C11 standard does require a guarantee, but gcc/g++ don't implement it I believe. So, writing in the source code that something has a guarantee may be misleading. Furthermore, the memory being overwritten is not the only thing we're after. The compiler is free to for example copy some of the memory to the stack first, where it could live longer.

However, if we restrict ourselves to supported systems and platforms, we know more at this point in time. We know what optimizations are implemented, and what compilers are capable of. It is reasonable that a call won't be optimized out if it's in another module, but that may not be the case if we'd switch to LTO builds, for example.